Client/App/v8world/Primitive.cpp

```diff
@@ -1 +1,439 @@
-#include "v8world/Primitive.h"
\ No newline at end of file
+#include "v8world/Primitive.h"
+#include "v8world/Geometry.h"
+#include "v8world/Assembly.h"
+#include "v8world/Ball.h"
+#include "v8world/Block.h"
+#include "v8world/World.h"
+#include <cmath>
+
+namespace RBX 
+{
+
+	Primitive::Primitive(Geometry::GeometryType geometryType) : // WIP
+		guidSetExternally(false),
+		world(NULL),
+		clump(NULL),
+		spatialNodes(NULL),
+		worldIndex(-1),
+		clumpDepth(-1),
+		traverseId(-1),
+		fuzzyExtentsStateId(-2),
+		geometry(newGeometry(geometryType)),
+		body(new Body),
+		elasticity(0.75f),
+		canCollide(true),
+		canSleep(true),
+		JointK(this, &Primitive::computeJointK)
+	{ // TODO: declare NullController class in controller.h
+		for(int i = 0; i < 6; i++) 
+		{
+			surfaceType[i] = NO_SURFACE;
+			surfaceData[i] = NULL;
+		}
+	}
+
+	Primitive::~Primitive() // WIP
+	{
+		Geometry::GeometryType type = geometry->getGeometryType();
+		if(type != Geometry::GEOMETRY_NONE) {
+			if(geometry) 
+				delete geometry;
+			if(body)
+				delete body;
+		}
+		for(int i = 0; i < 6; i++)
+			delete surfaceData[i];
+
+		RBXASSERT(!inPipeline());
+		RBXASSERT(!clump && !world);
+	}
+
+	G3D::Vector3 Primitive::clipToSafeSize(const G3D::Vector3& newSize)
+	{
+		G3D::Vector3 r = newSize.min(Vector3(512.0f, 512.0f, 512.0f));
+
+		if(r.x * r.y * r.z > 1e+6f) 
+		{
+			r.y = floorf(1e+6f / (r.x * r.z));
+		}
+
+		return r;
+	}
+
+	float Primitive::computeJointK() const 
+	{
+		Geometry::GeometryType type = geometry->getGeometryType();
+		return Constants::getJointK(geometry->getGridSize(), type == Geometry::GEOMETRY_BALL);
+	}
+
+	void Primitive::setGuid(const Guid& value) 
+	{
+		RBXASSERT(world);
+		guid.copyDataFrom(value);
+		guidSetExternally = true;
+	}
+
+	void Primitive::setClump(Clump* set)
+	{
+		if(set != clump)
+		{
+			clump = set;
+		}
+	}
+
+	const G3D::CoordinateFrame& Primitive::getCoordinateFrame() const
```

Comment:
This function seems to be inlining in some functions but not in others which causes issues matching them (could it be 2 separate functions?) to workaround this you could make the main function noinline and then make another identical function with inlining and use whichever one is needed